recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +56,89 @@ def validate(self):
         use_windows_commands = os.name == 'nt'
         command = "build" if use_windows_commands else "./build.sh"
         if self.options.toolset != 'auto':
             command += " "+str(self.options.toolset)
-        build_dir = os.path.join(self.source_folder, "source")
-        engine_dir = os.path.join(build_dir, "src", "engine")
-        os.chdir(engine_dir)
-        with tools.environment_append({"VSCMD_START_DIR": os.curdir}):
-            if self.options.use_cxx_env:
-                # Allow use of CXX env vars.
+        with conan.tools.files.chdir(self, self.b2_engine_dir):
+            with self.bootstrap_env():
                 self.run(command)
-            else:
-                # To avoid using the CXX env vars we clear them out for the build.
-                with tools.environment_append({"CXX": "", "CXXFLAGS": ""}):
-                    self.run(command)
-        os.chdir(build_dir)
+
+        self.output.info("Install..")
         command = os.path.join(
-            engine_dir, "b2.exe" if use_windows_commands else "b2")
+            self.b2_engine_dir, "b2.exe" if use_windows_commands else "b2")
         # auto, cxx, and cross-cxx aren't toolsets in b2; they're only used to affect
         # the way build.sh builds b2. Don't pass them to b2 itself.
         if self.options.toolset not in ['auto', 'cxx', 'cross-cxx']:
             command += " toolset=" + str(self.options.toolset)
         full_command = \
-            "{0} --ignore-site-config --prefix=../output --abbreviate-paths install b2-install-layout=portable".format(
-                command)
-        self.run(full_command)
+            ("{0} --ignore-site-config " +
+             "--prefix={1} " +
+             "--abbreviate-paths " +
+             "install " +
+             "b2-install-layout=portable").format(command, self.b2_output_dir)
+        with conan.tools.files.chdir(self, self.b2_dir):
+            self.run(full_command)
 
     def package(self):
-        self.copy("LICENSE.txt", dst="licenses", src="source")
-        self.copy(pattern="*b2", dst="bin", src="output")
-        self.copy(pattern="*b2.exe", dst="bin", src="output")
-        self.copy(pattern="*.jam", dst="bin", src="output")
+        conan.tools.files.copy(
+            self, "LICENSE.txt", dst=self.pkg_licenses_dir, src=self.source_folder)
+        conan.tools.files.copy(
+            self, "*b2", dst=self.pkg_bin_dir, src=self.b2_output_dir)
+        conan.tools.files.copy(
+            self, "*b2.exe", dst=self.pkg_bin_dir, src=self.b2_output_dir)
+        conan.tools.files.copy(
+            self, "*.jam", dst=self.pkg_bin_dir, src=self.b2_output_dir)
 
     def package_info(self):
         self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
         self.cpp_info.bindirs = ["bin"]
-        self.env_info.path = [os.path.join(
-            self.package_folder, "bin")]
+        self.buildenv_info.prepend_path("PATH", self.pkg_bin_dir)
```

Comment:
```suggestion
```

Not required. `VirtualBuildEnv` already does it automatically:

https://docs.conan.io/en/latest/reference/conanfile/tools/env/virtualbuildenv.html

> conanbuildenv-release-x86_64.(bat|sh): This file contains the actual definition of environment variables like PATH, LD_LIBRARY_PATH, etc, and any other variable defined in the dependencies buildenv_info corresponding to the build contex